dbt_platform_helper/utils/validation.py

```diff
@@ -557,8 +563,98 @@ def validate_platform_config(config, disable_aws_validation=False):
     _validate_environment_pipelines_triggers(enriched_config)
     _validate_codebase_pipelines(enriched_config)
     validate_database_copy_section(enriched_config)
+
     if not disable_aws_validation:
         _validate_s3_bucket_uniqueness(enriched_config)
+        _validate_redis_versions(config)
+        _validate_opensearch_versions(config)
+
+
+def _validate_redis_versions(config):
+
+    supported_redis_versions = []
+
+    if cache_refresh_required("redis"):
+        supported_redis_versions = get_redis_supported_versions()
+    else:
+        supported_redis_versions = read_supported_versions_from_cache("redis")
+
+    redis_extensions = []
+    redis_environments_with_failure = []
+
+    for extension in config.get("extensions").keys():
+
+        if config.get("extensions").get(extension).get("type") == "redis":
+            redis_extensions.append(config["extensions"][extension])
+
+    for redis_extension in redis_extensions:
+
+        environments = redis_extension.get("environments", {})
+
+        if not isinstance(environments, dict):
+            click.secho(
+                "Error: Redis extension definition is invalid type, expected dictionary", fg="red"
+            )
+            continue
+
+        for environment, env_config in environments.items():
+
```

Comment:
I would prefer not to have these redundant line breaks, like we don't about 8 lines down.